sdk/tests/tutorials/ibor/test_reference_portfolios.py

```diff
@@ -1,33 +1,55 @@
 import unittest
-import uuid
+import pytz
+import json
+import logging
+
+from datetime import datetime
 
 import lusid
 import lusid.models as models
 
 from lusidfeature import lusid_feature
 from utilities import TestDataUtilities
+from utilities import InstrumentLoader
 
 
 class ReferencePortfolio(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
 
+        # Log any exceptions
+        cls.logger = logging.getLogger()
+        cls.logger.setLevel(logging.INFO)
+
         # Create API client
         api_client = TestDataUtilities.api_client()
 
-        # Instantiate reference portfolio and portfolios API
+        # Instantiate APIs we will use
         cls.reference_portfolio_api = lusid.ReferencePortfolioApi(api_client)
         cls.portfolios_api = lusid.PortfoliosApi(api_client)
+        cls.instruments_api = lusid.InstrumentsApi(api_client)
+
+        # Load instruments
+        cls.instrument_loader = InstrumentLoader(cls.instruments_api)
+        cls.instrument_ids = cls.instrument_loader.load_instruments()
+
+        cls.reference_portfolio_name = "Test Reference Portfolio name"
```

Comment:
move these into each of the examples and delete in there, there is a (very small) chance that if the tests run at the same time you will get a clash

Reply:
try to avoid using test, demo in names

Reply:
Hey @rizwansaeed, do you mean delete the portfolio in each of the tests instead of using `tearDown`?

Reply:
yeah, I don't know if  you can parameterise the `tearDown`